pyproject.toml

```diff
@@ -2,11 +2,70 @@
 requires = ["setuptools>=45", "wheel", "setuptools_scm>=8.1.0"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "PyMeasure"
+# version is added dynamically
+authors = [
+    {name = "PyMeasure Developers"}
+]
+description = "Scientific measurement library for instruments, experiments, and live-plotting"
+# readme is merged dynamically from README and changelog
+requires-python = ">=3.8"
+keywords = ["measure", "instrument", "experiment control", "automate", "graph", "plot"]
+license = {text = "MIT"}
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Science/Research",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: MacOS",
+    "Operating System :: Microsoft :: Windows",
+    "Operating System :: POSIX",
+    "Operating System :: Unix",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+    "Topic :: Scientific/Engineering",
+]
+dependencies = [
+    "numpy>=1.6.1,<3",
+    "pandas>=0.14,<3",
+    "pint",
+    "pyvisa>=1.9",
+    "pyserial>=2.7",
+    "pyqtgraph>=0.12",
+]
+dynamic = ["version", "readme"]
+
+[project.optional-dependencies]
+tcp = ["pyzmq>=16.0.2", "cloudpickle>=0.3.1"]
+python-vxi11 = ["python-vxi11>=0.9"]
+tests = ["pytest>=3.3.0", "pytest-cov>=4.1.0", "pytest-qt>=2.4.0", "pyvisa-sim>=0.4.0"]
+# install pyqt or pyside manually as desired
+docs = ["sphinx>=5.0.0", "sphinx_rtd_theme>=1.1.0"]
```

Comment:
These dependencies should be one per line, in order to make version changes easier to spot in git.